src/shared/Core/Diagnostics/EnvironmentDiagnostic.cs

```diff
@@ -10,8 +11,8 @@ public class EnvironmentDiagnostic : Diagnostic
     {
         private readonly IEnvironment _env;
 
-        public EnvironmentDiagnostic(IEnvironment env)
-            : base("Environment")
+        public EnvironmentDiagnostic(IEnvironment env, ITrace2 trace2)
```

Comment:
Given that the various `Diagnostics` need many of the components from `ICommandContext`, maybe we should just always pass the entire `ICommandContext` to each one?